### DISCUSSION TITLE
Integrate non OpenAI model (ex Llama3) deployed on Azure

### DISCUSSION THREAD

**Discussion by vmastandrea-fincons** [langchain4j/langchain4j#1580] [Q&A]
Hi everyone, and thanks to those who can help me!

I'm trying to use LangChain4j to integrate a Llama3 model, deployed on Azure, into my Spring Boot application. However, it seems that the instructions provided in the guide are only applicable to OpenAI's LLM on Azure.

I attempted to modify the URL and API key, but when I use a model other than OpenAI, I encounter the following error: 'Error generating response: {status=Auth token must be passed as a header called Authorization}.'

Is there a way to achieve my goal?"

**Comment by agoncal** [comment]
I've deployed a Lllama 3 model to Azure OpenAI:

![Screenshot 2024-10-24 at 11 34 12](https://github.com/user-attachments/assets/16b022b1-ebde-4aa9-8985-174c5fcef181)

And when I try to access it with the API key and Endoint:

```java
AzureOpenAiChatModel model = AzureOpenAiChatModel.builder()
  .apiKey(AZURE_OPENAI_KEY)
  .endpoint(AZURE_OPENAI_ENDPOINT)
  .deploymentName("llama-3-2-1b-instruct-1")
  .logRequestsAndResponses(true)
  .build();

System.out.println(model.generate("When was the first Rolling Stones album released?"));
```

I get a:

```
Status code 403, "key_auth_bad_header_forbidden
```

I'm investigating: https://learn.microsoft.com/en-us/answers/questions/2110416/accessing-a-llama-3-model-deployed-on-azure-openai

**Reply by jdubois** [reply]
Be careful, as European citizens I believe we're not allowed to deploy this model

**Reply by qkfang** [reply]
My feeling is that it is unlikely to work. The llama-3-2-1b-instruct-1 (any open source LLM) is deployed to *.inference.ml.azure.com that is different to the GPT models deployment as a part of Azure OpenAI service.

Guess you used the target uri from the deployment endpoint in the format of: https://xxxx.eastus2.inference.ml.azure.com. if you compare this with your working GPT-4o's endpoint url, you will notice the difference.

On the other hand, the GPT-4o's request structure would be slightly different to llama 3.2's. the error you see is likely related to the authentication. If you go to Consume tab, and look at the example python code, the required auth header is a bearer header, but the AzureOpenAiChatModel would be using a api-key header.

In short, dont believe the AzureOpenAiChatModel support llama-3-2-1b-instruct-1 model out of the box yet.
